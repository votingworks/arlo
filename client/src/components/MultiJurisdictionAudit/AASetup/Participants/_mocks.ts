import { readFileSync } from 'fs'
import { join } from 'path'

const jurisdictionFile = new File(
  [
    readFileSync(
      join(
        __dirname,
        '../../../../../public/sample_jurisdiction_filesheet.csv'
      ),
      'utf8'
    ),
  ],
  'jurisdictions.csv',
  { type: 'text/csv' }
)

export default jurisdictionFile
