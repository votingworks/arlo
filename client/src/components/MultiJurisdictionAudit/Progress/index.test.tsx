import React from 'react'
import { render, screen, within } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Progress from '.'
import { jurisdictionMocks, auditSettings } from '../useSetupMenuItems/_mocks'

jest.mock('react-router', () => ({
  useParams: jest.fn().mockReturnValue({ electionId: '1' }),
}))

describe('Progress screen', () => {
  it('shows ballot manifest upload status', () => {
    const { container } = render(
      <Progress
        jurisdictions={jurisdictionMocks.oneManifest}
        auditSettings={auditSettings.all}
      />
    )

    screen.getByText('Audit Progress by Jurisdiction')
    screen.getByRole('columnheader', { name: 'Jurisdiction Name' })
    screen.getByRole('columnheader', { name: 'Status' })
    screen.getByRole('columnheader', { name: 'Total Audited' })
    screen.getByRole('columnheader', { name: 'Remaining in Round' })
    const rows = screen.getAllByRole('row')
    expect(rows).toHaveLength(jurisdictionMocks.oneManifest.length + 1) // includes headers
    within(rows[1]).getByRole('cell', { name: 'Jurisdiction 1' })
    within(rows[1]).getByRole('cell', {
      name: 'Manifest upload failed: Invalid CSV',
    })
    within(rows[2]).getByRole('cell', { name: 'Jurisdiction 2' })
    within(rows[2]).getByRole('cell', { name: 'No manifest uploaded' })
    within(rows[3]).getByRole('cell', { name: 'Jurisdiction 3' })
    within(rows[3]).getByRole('cell', { name: 'Manifest received' })
    expect(container).toMatchSnapshot()
  })

  it('shows round status', () => {
    const { container } = render(
      <Progress
        jurisdictions={jurisdictionMocks.oneComplete}
        auditSettings={auditSettings.all}
      />
    )

    screen.getByText('Audit Progress by Jurisdiction')
    screen.getByRole('columnheader', { name: 'Jurisdiction Name' })
    screen.getByRole('columnheader', { name: 'Status' })
    screen.getByRole('columnheader', { name: 'Total Audited' })
    screen.getByRole('columnheader', { name: 'Remaining in Round' })
    const rows = screen.getAllByRole('row')
    expect(rows).toHaveLength(jurisdictionMocks.oneManifest.length + 1) // includes headers
    within(rows[1]).getByRole('cell', { name: 'Jurisdiction 1' })
    within(rows[1]).getByRole('cell', { name: 'In progress' })
    within(rows[1]).getByRole('cell', { name: '4' })
    within(rows[1]).getByRole('cell', { name: '6' })
    within(rows[2]).getByRole('cell', { name: 'Jurisdiction 2' })
    within(rows[2]).getByRole('cell', { name: 'Not started' })
    within(rows[2]).getByRole('cell', { name: '0' })
    within(rows[2]).getByRole('cell', { name: '20' })
    within(rows[3]).getByRole('cell', { name: 'Jurisdiction 3' })
    within(rows[3]).getByRole('cell', { name: 'Complete' })
    within(rows[3]).getByRole('cell', { name: '30' })
    within(rows[3]).getByRole('cell', { name: '0' })
    expect(container).toMatchSnapshot()
  })

  it('toggles between ballots and samples', () => {
    render(
      <Progress
        jurisdictions={jurisdictionMocks.oneComplete}
        auditSettings={auditSettings.all}
      />
    )

    const ballotsSwitch = screen.getByRole('checkbox', {
      name: 'Count unique sampled ballots',
    })
    userEvent.click(ballotsSwitch)
    let rows = screen.getAllByRole('row')
    within(rows[1]).getByRole('cell', { name: '5' })
    within(rows[1]).getByRole('cell', { name: '6' })
    within(rows[2]).getByRole('cell', { name: '0' })
    within(rows[2]).getByRole('cell', { name: '22' })
    within(rows[3]).getByRole('cell', { name: '31' })
    within(rows[3]).getByRole('cell', { name: '0' })

    userEvent.click(ballotsSwitch)
    rows = screen.getAllByRole('row')
    within(rows[1]).getByRole('cell', { name: '4' })
    within(rows[1]).getByRole('cell', { name: '6' })
    within(rows[2]).getByRole('cell', { name: '0' })
    within(rows[2]).getByRole('cell', { name: '20' })
    within(rows[3]).getByRole('cell', { name: '30' })
    within(rows[3]).getByRole('cell', { name: '0' })
  })

  it('shows a different toggle label for batch audits', () => {
    render(
      <Progress
        jurisdictions={jurisdictionMocks.oneComplete}
        auditSettings={auditSettings.batchComparisonAll}
      />
    )
    screen.getByRole('checkbox', {
      name: 'Count unique sampled batches',
    })
  })

  it('shows the detail modal', () => {
    const { container } = render(
      <Progress
        jurisdictions={jurisdictionMocks.oneManifest}
        auditSettings={auditSettings.all}
      />
    )

    // Click on a jurisdiction name to open the detail modal
    userEvent.click(screen.getByRole('button', { name: 'Jurisdiction 1' }))
    screen.getByText('Jurisdiction 1 Audit Information')
    expect(container).toMatchSnapshot()

    // Close the detail modal
    userEvent.click(screen.getAllByRole('button', { name: 'Close' })[0])
    expect(
      screen.queryByText('Jurisdiction 1 Audit Information')
    ).not.toBeInTheDocument()
  })

  it('shows the detail modal for a jurisdiction without an upload', () => {
    const { container } = render(
      <Progress
        jurisdictions={jurisdictionMocks.oneManifest}
        auditSettings={auditSettings.all}
      />
    )

    // Click on a jurisdiction name to open the detail modal
    userEvent.click(screen.getByRole('button', { name: 'Jurisdiction 2' }))
    screen.getByText('Jurisdiction 2 Audit Information')
    expect(container).toMatchSnapshot()

    // Close the detail modal
    userEvent.click(screen.getAllByRole('button', { name: 'Close' })[0])
    expect(
      screen.queryByText('Jurisdiction 2 Audit Information')
    ).not.toBeInTheDocument()
  })

  it('filters by jurisdiction name', async () => {
    render(
      <Progress
        jurisdictions={jurisdictionMocks.oneManifest}
        auditSettings={auditSettings.all}
      />
    )

    const filter = screen.getByPlaceholderText('Filter by jurisdiction name...')
    await userEvent.type(filter, '1')
    expect(screen.getAllByRole('row')).toHaveLength(1 + 1) // includes headers
    screen.getByRole('cell', { name: 'Jurisdiction 1' })

    userEvent.clear(filter)
    await userEvent.type(filter, 'Jurisdiction')
    expect(screen.getAllByRole('row')).toHaveLength(
      jurisdictionMocks.oneManifest.length + 1
    )
  })

  it('sorts', () => {
    const { rerender } = render(
      <Progress
        jurisdictions={jurisdictionMocks.oneManifest}
        auditSettings={auditSettings.all}
      />
    )

    // Toggle sorting by name
    // First click doesn't change order because they are sorted by name by default
    const nameHeader = screen.getByRole('columnheader', {
      name: 'Jurisdiction Name',
    })
    userEvent.click(nameHeader)
    let rows = screen.getAllByRole('row')
    within(rows[1]).getByRole('cell', { name: 'Jurisdiction 1' })

    userEvent.click(nameHeader)
    rows = screen.getAllByRole('row')
    within(rows[1]).getByRole('cell', { name: 'Jurisdiction 3' })

    userEvent.click(nameHeader)
    rows = screen.getAllByRole('row')
    within(rows[1]).getByRole('cell', { name: 'Jurisdiction 1' })

    // Toggle sorting by status
    let statusHeader = screen.getByRole('columnheader', {
      name: 'Status',
    })
    userEvent.click(statusHeader)
    rows = screen.getAllByRole('row')
    within(rows[1]).getByRole('cell', { name: 'No manifest uploaded' })

    userEvent.click(statusHeader)
    rows = screen.getAllByRole('row')
    within(rows[1]).getByRole('cell', { name: 'Manifest received' })

    userEvent.click(statusHeader)
    rows = screen.getAllByRole('row')
    within(rows[1]).getByRole('cell', {
      name: 'Manifest upload failed: Invalid CSV',
    })

    // Toggle sorting by status once audit begins
    rerender(
      <Progress
        jurisdictions={jurisdictionMocks.oneComplete}
        auditSettings={auditSettings.all}
      />
    )

    statusHeader = screen.getByRole('columnheader', {
      name: 'Status',
    })
    userEvent.click(statusHeader)
    rows = screen.getAllByRole('row')
    within(rows[1]).getByRole('cell', { name: 'Not started' })

    userEvent.click(statusHeader)
    rows = screen.getAllByRole('row')
    within(rows[1]).getByRole('cell', { name: 'Complete' })

    userEvent.click(statusHeader)
    rows = screen.getAllByRole('row')
    within(rows[1]).getByRole('cell', { name: 'In progress' })
  })
})
