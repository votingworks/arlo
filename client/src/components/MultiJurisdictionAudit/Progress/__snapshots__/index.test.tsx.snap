// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress screen shows ballot manifest upload status 1`] = `
<div>
  <div
    class="sc-ifAKCX iABumv"
  >
    <h2
      class="bp3-heading sc-bdVaJa ibVpPq"
    >
      Audit Progress by Jurisdiction
    </h2>
    <p>
      Click on a column name to sort by that column's data. To reverse sort, click on the column name again.
      <br />
       To view a single jurisdiction's data, click the name of the jurisdiction.
    </p>
    <div
      class="sc-EHOje enakgn"
    >
      <label
        class="bp3-control bp3-switch"
      >
        <input
          checked=""
          type="checkbox"
        />
        <span
          class="bp3-control-indicator"
        />
        Count unique sampled ballots
      </label>
      <div
        class="bp3-input-group .modifier"
      >
        <span
          class="bp3-icon bp3-icon-filter"
        />
        <input
          class="bp3-input"
          placeholder="Filter by jurisdiction name..."
          type="text"
          value=""
        />
      </div>
    </div>
    <table
      class="sc-bxivhb eaTstv"
      role="table"
    >
      <thead>
        <tr>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Jurisdiction Name"
          >
            Jurisdiction Name
            <span>
              <span
                class="bp3-icon bp3-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Status"
          >
            Status
            <span>
              <span
                class="bp3-icon bp3-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Total Audited"
          >
            Total Audited
            <span>
              <span
                class="bp3-icon bp3-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Remaining in Round"
          >
            Remaining in Round
            <span>
              <span
                class="bp3-icon bp3-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <button
              class="bp3-button bp3-minimal bp3-small bp3-intent-primary"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                Jurisdiction 1
              </span>
            </button>
          </td>
          <td
            role="cell"
          >
            Manifest upload failed: Invalid CSV
          </td>
          <td
            role="cell"
          />
          <td
            role="cell"
          />
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <button
              class="bp3-button bp3-minimal bp3-small bp3-intent-primary"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                Jurisdiction 2
              </span>
            </button>
          </td>
          <td
            role="cell"
          >
            No manifest uploaded
          </td>
          <td
            role="cell"
          />
          <td
            role="cell"
          />
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <button
              class="bp3-button bp3-minimal bp3-small bp3-intent-primary"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                Jurisdiction 3
              </span>
            </button>
          </td>
          <td
            role="cell"
          >
            Manifest received
          </td>
          <td
            role="cell"
          />
          <td
            role="cell"
          />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Progress screen shows round status 1`] = `
<div>
  <div
    class="sc-ifAKCX iABumv"
  >
    <h2
      class="bp3-heading sc-bdVaJa ibVpPq"
    >
      Audit Progress by Jurisdiction
    </h2>
    <p>
      Click on a column name to sort by that column's data. To reverse sort, click on the column name again.
      <br />
       To view a single jurisdiction's data, click the name of the jurisdiction.
    </p>
    <div
      class="sc-EHOje enakgn"
    >
      <label
        class="bp3-control bp3-switch"
      >
        <input
          checked=""
          type="checkbox"
        />
        <span
          class="bp3-control-indicator"
        />
        Count unique sampled ballots
      </label>
      <div
        class="bp3-input-group .modifier"
      >
        <span
          class="bp3-icon bp3-icon-filter"
        />
        <input
          class="bp3-input"
          placeholder="Filter by jurisdiction name..."
          type="text"
          value=""
        />
      </div>
    </div>
    <table
      class="sc-bxivhb eaTstv"
      role="table"
    >
      <thead>
        <tr>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Jurisdiction Name"
          >
            Jurisdiction Name
            <span>
              <span
                class="bp3-icon bp3-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Status"
          >
            Status
            <span>
              <span
                class="bp3-icon bp3-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Total Audited"
          >
            Total Audited
            <span>
              <span
                class="bp3-icon bp3-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Remaining in Round"
          >
            Remaining in Round
            <span>
              <span
                class="bp3-icon bp3-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <button
              class="bp3-button bp3-minimal bp3-small bp3-intent-primary"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                Jurisdiction 1
              </span>
            </button>
          </td>
          <td
            role="cell"
          >
            In progress
          </td>
          <td
            role="cell"
          >
            4
          </td>
          <td
            role="cell"
          >
            6
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <button
              class="bp3-button bp3-minimal bp3-small bp3-intent-primary"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                Jurisdiction 2
              </span>
            </button>
          </td>
          <td
            role="cell"
          >
            Not started
          </td>
          <td
            role="cell"
          >
            0
          </td>
          <td
            role="cell"
          >
            20
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <button
              class="bp3-button bp3-minimal bp3-small bp3-intent-primary"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                Jurisdiction 3
              </span>
            </button>
          </td>
          <td
            role="cell"
          >
            Complete
          </td>
          <td
            role="cell"
          >
            30
          </td>
          <td
            role="cell"
          >
            0
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Progress screen shows the detail modal 1`] = `
<div>
  <div
    class="sc-ifAKCX iABumv"
  >
    <h2
      class="bp3-heading sc-bdVaJa ibVpPq"
    >
      Audit Progress by Jurisdiction
    </h2>
    <p>
      Click on a column name to sort by that column's data. To reverse sort, click on the column name again.
      <br />
       To view a single jurisdiction's data, click the name of the jurisdiction.
    </p>
    <div
      class="sc-EHOje enakgn"
    >
      <label
        class="bp3-control bp3-switch"
      >
        <input
          checked=""
          type="checkbox"
        />
        <span
          class="bp3-control-indicator"
        />
        Count unique sampled ballots
      </label>
      <div
        class="bp3-input-group .modifier"
      >
        <span
          class="bp3-icon bp3-icon-filter"
        />
        <input
          class="bp3-input"
          placeholder="Filter by jurisdiction name..."
          type="text"
          value=""
        />
      </div>
    </div>
    <table
      class="sc-bxivhb eaTstv"
      role="table"
    >
      <thead>
        <tr>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Jurisdiction Name"
          >
            Jurisdiction Name
            <span>
              <span
                class="bp3-icon bp3-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Status"
          >
            Status
            <span>
              <span
                class="bp3-icon bp3-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Total Audited"
          >
            Total Audited
            <span>
              <span
                class="bp3-icon bp3-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Remaining in Round"
          >
            Remaining in Round
            <span>
              <span
                class="bp3-icon bp3-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <button
              class="bp3-button bp3-minimal bp3-small bp3-intent-primary"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                Jurisdiction 1
              </span>
            </button>
          </td>
          <td
            role="cell"
          >
            Manifest upload failed: Invalid CSV
          </td>
          <td
            role="cell"
          />
          <td
            role="cell"
          />
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <button
              class="bp3-button bp3-minimal bp3-small bp3-intent-primary"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                Jurisdiction 2
              </span>
            </button>
          </td>
          <td
            role="cell"
          >
            No manifest uploaded
          </td>
          <td
            role="cell"
          />
          <td
            role="cell"
          />
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <button
              class="bp3-button bp3-minimal bp3-small bp3-intent-primary"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                Jurisdiction 3
              </span>
            </button>
          </td>
          <td
            role="cell"
          >
            Manifest received
          </td>
          <td
            role="cell"
          />
          <td
            role="cell"
          />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Progress screen shows the detail modal for a jurisdiction without an upload 1`] = `
<div>
  <div
    class="sc-ifAKCX iABumv"
  >
    <h2
      class="bp3-heading sc-bdVaJa ibVpPq"
    >
      Audit Progress by Jurisdiction
    </h2>
    <p>
      Click on a column name to sort by that column's data. To reverse sort, click on the column name again.
      <br />
       To view a single jurisdiction's data, click the name of the jurisdiction.
    </p>
    <div
      class="sc-EHOje enakgn"
    >
      <label
        class="bp3-control bp3-switch"
      >
        <input
          checked=""
          type="checkbox"
        />
        <span
          class="bp3-control-indicator"
        />
        Count unique sampled ballots
      </label>
      <div
        class="bp3-input-group .modifier"
      >
        <span
          class="bp3-icon bp3-icon-filter"
        />
        <input
          class="bp3-input"
          placeholder="Filter by jurisdiction name..."
          type="text"
          value=""
        />
      </div>
    </div>
    <table
      class="sc-bxivhb eaTstv"
      role="table"
    >
      <thead>
        <tr>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Jurisdiction Name"
          >
            Jurisdiction Name
            <span>
              <span
                class="bp3-icon bp3-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Status"
          >
            Status
            <span>
              <span
                class="bp3-icon bp3-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Total Audited"
          >
            Total Audited
            <span>
              <span
                class="bp3-icon bp3-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Remaining in Round"
          >
            Remaining in Round
            <span>
              <span
                class="bp3-icon bp3-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <button
              class="bp3-button bp3-minimal bp3-small bp3-intent-primary"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                Jurisdiction 1
              </span>
            </button>
          </td>
          <td
            role="cell"
          >
            Manifest upload failed: Invalid CSV
          </td>
          <td
            role="cell"
          />
          <td
            role="cell"
          />
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <button
              class="bp3-button bp3-minimal bp3-small bp3-intent-primary"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                Jurisdiction 2
              </span>
            </button>
          </td>
          <td
            role="cell"
          >
            No manifest uploaded
          </td>
          <td
            role="cell"
          />
          <td
            role="cell"
          />
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <button
              class="bp3-button bp3-minimal bp3-small bp3-intent-primary"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                Jurisdiction 3
              </span>
            </button>
          </td>
          <td
            role="cell"
          >
            Manifest received
          </td>
          <td
            role="cell"
          />
          <td
            role="cell"
          />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
